File: CalystoShared/Calysto.Framework/Calysto.Web/WebLib/Engine/Src/Core/Net/Infrastructure/IWebServiceResponseContainer.ts
namespace Calysto.Net.WebService
{
	/** istoimeni tip postoji u C# */
	export interface IWebServiceResponseContainer
	{
		/** true if server method invoked without exception thrown, returning void or value */
		IsSuccessful?: boolean;

		/** "IWebServiceResponseContainer" */
		Type?: "IWebServiceResponseContainer";

		/** */
		IsEngineExpired?: boolean;

		////JavaScriptHI?: string; // not used any more

		/**execute javascript */
		JavaScriptLO?: string;

		/** once the JavaScriptLO is used, mark it as done to prevent double execution */
		IsJavaScriptLODone: boolean;

		/** short message */
		ExceptionMessage?: string;

		/** once the ExceptionMessage is used, mark it as done to prevent double execution */
		IsExceptionMessageDone: boolean;

		/** detailed message */
		ExceptionDetails?: string;

		/** method name on server */
		Method?: string;

		/**unique message reference */
		ReferenceGuid: string;

		/** socket echo message */
		EchoMsg: string;
	}
}

File: CalystoShared/Calysto.Framework/Calysto.Genesis/Calysto/Web/Script/Services/IWebServiceResponseContainer.cs
namespace Calysto.Web.Script.Services
{
	public interface IWebServiceResponseContainer
	{
		bool IsSuccessful { get; set; }

		/** may have value "IWebServiceResponseContainer" */
		string Type { get; set; }

		bool IsEngineExpired { get; set; }

		string JavaScriptLO { get; set; }

		/** short message */
		string ExceptionMessage { get; set; }

		/** detailed message */
		string ExceptionDetails { get; set; }

		object ReturnedValue { get; set; }

		/** method name on server */
		string Method { get; set; }

		/**unique message reference*/
		string ReferenceGuid { get; set; }

		/** socket echo message */
		string EchoMsg { get; set; }
	};

	public interface IWebServiceResponseContainerWithReturn<TReturn>: IWebServiceResponseContainer
	{
		/** value returned from Method on server */
		new TReturn ReturnedValue { get; set; }
	};
}